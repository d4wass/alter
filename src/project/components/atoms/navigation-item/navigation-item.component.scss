a {
  text-decoration: none;
  color: #000;
}
.active-link {
  font-weight: bold;
}
