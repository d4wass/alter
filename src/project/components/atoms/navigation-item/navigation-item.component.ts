import { Component, Input } from '@angular/core';
import { urlItem } from 'src/router/routes';

@Component({
  selector: 'app-navigation-item',
  template: `
    <a
      [routerLink]="['/', item?.url]"
      routerLinkActive="active-link"
      data-test="link"
      class="link"
      >{{ item?.description }}</a
    >
  `,
  styleUrls: ['./navigation-item.component.scss']
})
export class NavigationItemComponent {
  @Input() item?: urlItem;
}
