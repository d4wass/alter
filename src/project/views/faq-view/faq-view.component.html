<app-title-header [title]="'How Alter works?'"></app-title-header>
<div class="flex-wrapper">
  <app-how-works-section></app-how-works-section>
  <app-cta-section></app-cta-section>
  <app-carousel-section></app-carousel-section>
  <app-benefit-section></app-benefit-section>
</div>
