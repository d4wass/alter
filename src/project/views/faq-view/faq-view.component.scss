// $gridGap: 2em;
// $mainSidePadding: 6.3em;

// .flex-wrapper {
//   padding: $gridGap $mainSidePadding;
// }
