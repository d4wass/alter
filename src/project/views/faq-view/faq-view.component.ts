import { Component } from '@angular/core';

@Component({
  selector: 'app-faq-view',
  templateUrl: './faq-view.component.html',
  styleUrls: ['./faq-view.component.scss']
})
export class FaqViewComponent {
  constructor() {}
}
