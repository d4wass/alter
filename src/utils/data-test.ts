export function dataTest(selector: string) {
  return `[data-test="${selector}"]`;
}
