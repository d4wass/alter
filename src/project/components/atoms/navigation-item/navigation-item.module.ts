import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NavigationItemComponent } from './navigation-item.component';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [NavigationItemComponent],
  imports: [CommonModule, RouterModule],
  exports: [NavigationItemComponent]
})
export class NavigationItemModule {}
