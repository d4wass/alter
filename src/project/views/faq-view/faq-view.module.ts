import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FaqViewComponent } from './faq-view.component';
import { TitleHeaderModule } from 'src/project/components/molecules/title-header/title-header.module';
import { HowWorksSectionModule } from 'src/project/components/molecules/how-works-section/how-works-section.module';
import { CtaSectionComponent } from 'src/project/components/molecules/cta-section/cta-section.component';
import { CarouselSectionModule } from 'src/project/components/organisms/carousel-section/carousel-section.module';
import { BenefitSectionModule } from 'src/project/components/molecules/benefit-section/benefit-section.module';

@NgModule({
  declarations: [FaqViewComponent, CtaSectionComponent],
  imports: [
    CommonModule,
    TitleHeaderModule,
    HowWorksSectionModule,
    CarouselSectionModule,
    BenefitSectionModule
  ]
})
export class FaqViewModule {}
