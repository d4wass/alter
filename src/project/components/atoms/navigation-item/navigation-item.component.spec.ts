import { createRoutingFactory, SpectatorRouting } from '@ngneat/spectator';
import { FaqViewComponent } from '../../../views/faq-view/faq-view.component';
import { dataTest } from '../../../../utils/data-test';

import { NavigationItemComponent } from './navigation-item.component';

describe('NavigationItemComponent', () => {
  let component: NavigationItemComponent;
  let spectator: SpectatorRouting<NavigationItemComponent>;

  const mock = {
    urlItem: { description: 'Learn more', url: 'faq' }
  };
  const selectors = {
    link: () => spectator.query(dataTest('link'))
  };

  const createComponent = createRoutingFactory({
    component: NavigationItemComponent,
    stubsEnabled: false,
    routes: [
      {
        path: '',
        component: NavigationItemComponent
      },
      { path: mock.urlItem.url, component: FaqViewComponent }
    ]
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('should match to snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should create component', () => {
    expect(component).toBeTruthy();
  });

  describe('(item)', () => {
    it('should set (item) to urlItem when gets value', () => {
      component.item = mock.urlItem;
      spectator.detectChanges();

      expect(component.item).toEqual(mock.urlItem);
      expect(selectors.link()?.innerHTML).toEqual(mock.urlItem.description);
    });
    it('should set (item) to undefined when not get value', () => {
      expect(component.item).toBeUndefined();
    });
  });

  describe('Router', () => {
    it('should change router on click when (item) has value', async () => {
      component.item = mock.urlItem;
      spectator.detectChanges();

      spectator.dispatchMouseEvent('.link', 'click');
      expect(spectator.router.url).toEqual(`/${mock.urlItem.url}`);
    });
    it('should not change router on click when (item) is undefined', () => {
      spectator.dispatchMouseEvent('.link', 'click');
      expect(spectator.router.url).toEqual(`/`);
    });
  });
});
