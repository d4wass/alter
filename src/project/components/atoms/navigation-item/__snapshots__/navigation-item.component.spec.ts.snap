// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItemComponent should match to snapshot 1`] = `
<app-navigation-item>
  <a
    class="link"
    data-test="link"
    href="/null"
    routerlinkactive="active-link"
  >
  </a>
</app-navigation-item>
`;
